chain/client/proof.go

```diff
@@ -408,155 +408,157 @@ func MakeOtherStoresMerkleHash(mspo rootmulti.MultiStoreProofOp) (cmn.HexBytes,
-		// if err != nil {
-		// 	panic(err)
-		// }
-
-		// blockRelayBytes, err := brp.encodeToEthData(uint64(rc.Height))
-		// if err != nil {
-		// 	rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
-		// }
-
-		// oracleDataBytes, err := odp.encodeToEthData(uint64(rc.Height))
-		// if err != nil {
-		// 	rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
-		// }
-
-		// evmProofBytes, err := relayAndVerifyArguments.Pack(blockRelayBytes, oracleDataBytes)
-		// if err != nil {
-		// 	rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
-		// }
-
-		// rest.PostProcessResponse(w, cliCtx, Proof{
-		// 	JsonProof: JsonProof{
-		// 		BlockHeight:     uint64(rc.Height),
-		// 		OracleDataProof: odp,
-		// 		BlockRelayProof: brp},
-		// 	EVMProofBytes: evmProofBytes,
-		// })
+		vars := mux.Vars(r)
+		reqIDStr := vars[requestID]
+		reqID, err := strconv.ParseUint(reqIDStr, 10, 64)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		rc, err := cliCtx.Client.Commit(nil)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		brp, err := GetBlockRelayProof(cliCtx, uint64(rc.Height))
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		var queryRequest zoracle.RequestQuerierInfo
+		res, _, err := cliCtx.QueryWithData(fmt.Sprintf("custom/zoracle/request/%d", reqID), nil)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		err = cliCtx.Codec.UnmarshalJSON(res, &queryRequest)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		key := zoracle.ResultStoreKey(int64(reqID), queryRequest.Request.OracleScriptID, queryRequest.Request.Calldata)
+
+		resp, err := cliCtx.Client.ABCIQueryWithOptions(
+			"/store/zoracle/key",
+			key,
+			rpcclient.ABCIQueryOptions{Height: rc.Height - 1, Prove: true},
+		)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		proof := resp.Response.GetProof()
+		if proof == nil {
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, "proof not found")
+			return
+		}
+
+		ops := proof.GetOps()
+		if ops == nil {
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, "proof ops not found")
+			return
+		}
+
+		var iavlOpData []byte
+		var multistoreOpData []byte
+		for _, op := range ops {
+			opType := op.GetType()
+			if opType == "iavl:v" {
+				iavlOpData = op.GetData()
+			} else if opType == "multistore" {
+				multistoreOpData = op.GetData()
+			}
+		}
+		if iavlOpData == nil || multistoreOpData == nil {
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, "proof was corrupted")
+			return
+		}
+
+		var opiavl iavl.IAVLValueOp
+		if iavlOpData == nil {
```

Comment:
If `iavlOpData==nil` code will return in line 478 already? why need to check here?